### PR TITLE
Adds bridge code for calling the stochastic sample method in RNA stru…

### PR DESCRIPTION
**Pull request by maxhwardg**
…cture.

### CODE REVIEW

File: src/bridge/rnastructure.h [hunk 1]
```diff
@@ -47,6 +51,26 @@ class RNAstructure : public RnaPackage {
   std::unique_ptr<datatable> data_;
   bool use_lyngso_;
 
+  struct PartitionFunctionState {
```
Comment (Edgeworth) [inline src/bridge/rnastructure.h:54]: Can you put this in an anonymous namespace in the implementation file?

File: src/bridge/rnastructure.cpp [hunk 2]
```diff
@@ -38,8 +39,45 @@ std::vector<Secondary> StructureToSecondarys(structure& struc) {
...
+    // TODO: Convert CTDs?
+    res.emplace_back(subopt::SuboptResult(
+        Energy(struc.GetEnergy(i + 1)), tb::TracebackResult(std::move(s_list[i]), Ctds())));
+  }
+  return res;
+}
+
 }  // namespace
 
+RNAstructure::PartitionFunctionState::PartitionFunctionState(int N, datatable* data) {
```
Comment (Edgeworth) [inline src/bridge/rnastructure.cpp:55]: Can you initialize these via constructor initializer list?

Comment (Edgeworth) [inline src/bridge/rnastructure.cpp:55]: For convention, could you rename this to PartitionState?

File: src/bridge/rnastructure.cpp [hunk 3]
```diff
@@ -93,57 +131,46 @@ std::vector<subopt::SuboptResult> RNAstructure::SuboptimalIntoVector(
...
-    res.emplace_back(subopt::SuboptResult(
-        Energy(structure->GetEnergy(i + 1)), tb::TracebackResult(std::move(s_list[i]), Ctds())));
-  }
-  return res;
+  return StructureToSuboptVector(*structure);
 }
 
 part::PartResult RNAstructure::Partition(const Primary& r) const {
   const auto structure = LoadStructure(r);
+  auto pf_state = CallPartitionFunction(structure.get());
```
Comment (Edgeworth) [inline src/bridge/rnastructure.cpp:139]: Here and other locations, could you call this just state?

File: src/bridge/rnastructure.cpp [hunk 4]
```diff
@@ -38,8 +39,45 @@ std::vector<Secondary> StructureToSecondarys(structure& struc) {
...
+  wcoax = make_unique<DynProgArray<PFPRECISION>>(N);
+  w5 = std::make_unique<PFPRECISION[]>(N + 1);
+  w3 = std::make_unique<PFPRECISION[]>(N + 2);
+  pfdata = std::make_unique<pfdatatable>(data, scaling, T);
+  fce = std::make_unique<forceclass>(N);
+  lfce = std::make_unique<bool[]>(2 * N + 1);
+  mod = std::make_unique<bool[]>(2 * N + 1);
+}
+
+RNAstructure::PartitionFunctionState RNAstructure::CallPartitionFunction(structure* struc) const {
```
Comment (Edgeworth) [inline src/bridge/rnastructure.cpp:71]: Could you call this RunPartition?

**Review by Edgeworth — CHANGES_REQUESTED** [review]

File: src/bridge/rnastructure.h (hunk 1 above)
Comment (maxhwardg) [reply src/bridge/rnastructure.h]: Done.

File: src/bridge/rnastructure.cpp (hunk 2 above)
Comment (maxhwardg) [reply src/bridge/rnastructure.cpp]: Done.

Comment (maxhwardg) [reply src/bridge/rnastructure.cpp]: Done.

File: src/bridge/rnastructure.cpp (hunk 3 above)
Comment (maxhwardg) [reply src/bridge/rnastructure.cpp]: Done.

FTR I consider pr_state more readable and reasonably inline with convention (i.e. s_list, pfdata). Can you explain the reason "state" is preferred?

File: src/bridge/rnastructure.cpp (hunk 4 above)
Comment (maxhwardg) [reply src/bridge/rnastructure.cpp]: Done.

**Pull request merged**